impl_core/src/main/java/io/opencensus/implcore/stats/MetricUtils.java

```diff
@@ -85,6 +88,14 @@ static Type getType(Measure measure, Aggregation aggregation) {
     return labelValues;
   }
 
+  static Map<String, String> toStringAttachments(Map<String, AttachmentValue> attachments) {
```

Comment:
For the moment `Exemplar`s in `Metric`s still have <string, stirng> attachments: https://github.com/census-instrumentation/opencensus-proto/blob/c9cc5de7251f8a320dd6cb4f92e5698ba86f284c/src/opencensus/proto/metrics/v1/metrics.proto#L263.

For now we need to convert `AttachmentValue` in Stats API to string in Metrics API.

Reply:
We can modify that because we respect the data model, the fact that we delay the string formatting doesn't mean we don't respect the proto. This is one of the reason we don't directly use the protos.

Reply:
That would be tricky since `AttachmentValue` is part of Stats APIs and Metrics APIs cannot have dependency on Stats APIs. If we really want to replace the <string, string> attachments in Metrics APIs, the only option I can see is to put `Exemplar` and `AttachmentValue` in common artifacts or a top-level API (which Go already does: https://github.com/census-instrumentation/opencensus-go/tree/master/exemplar).

Either way I would prefer to do it in a separate PR since this one is already large.

Reply:
Sure, agree, let's discuss in a separate PR.